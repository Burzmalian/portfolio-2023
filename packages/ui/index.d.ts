export * from "./src/Button/Button";
//# sourceMappingURL=index.d.ts.map