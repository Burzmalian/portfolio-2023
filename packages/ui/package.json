{
  "name": "ui",
  "version": "0.0.0",
  "main": "./index.tsx",
  "types": "./index.tsx",
  "license": "MIT",
  "scripts": {
    "dev": "start-storybook -p 6006",
    "build": "build-storybook --docs",
    "preview-storybook": "serve storybook-static",
    "clean": "rm -rf .turbo && rm -rf node_modules",
    "lint": "eslint *.ts*",
    "type-check": "tsc"
  },
  "dependencies": {
    "react": "^18.2.0",
    "react-dom": "^18.2.0"
  },
  "devDependencies": {
    "@types/react": "^18.0.17",
    "@types/react-dom": "^18.0.6",
    "eslint": "^7.32.0",
    "eslint-config-custom": "workspace:*",
    "tsconfig": "workspace:*",
    "typescript": "^4.5.2",
    "@storybook/addon-actions": "^6.5.14",
    "@storybook/addon-docs": "^6.5.14",
    "@storybook/addon-essentials": "^6.5.14",
    "@storybook/addon-links": "^6.5.14",
    "@storybook/builder-vite": "^0.1.41",
    "@storybook/react": "^6.5.14",
    "@vitejs/plugin-react": "^1.3.2",
    "serve": "^13.0.4",
    "vite": "^2.9.15"
  }
}
