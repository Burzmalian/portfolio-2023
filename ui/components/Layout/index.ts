export { default as Layout, LayoutParts } from "./Layout";
