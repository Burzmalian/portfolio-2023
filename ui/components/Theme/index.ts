export { ThemeProvider, useTheme } from "./Theme";
export { default as ThemeSwitch } from "./ThemeSwitch";
