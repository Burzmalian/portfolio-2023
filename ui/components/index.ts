export * from "./Actions/Button";
export * from "./Actions/Link";
export * from "./Layout";
export * from "./Theme";
export * from "./Header";
