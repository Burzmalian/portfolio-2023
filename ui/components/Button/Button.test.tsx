import { render, screen } from "@testing-library/react";
import { Button } from "@/ui/components";

describe("Button", () => {
  it("renders a button", () => {
    render(<Button />);

    const button = screen.getByTestId("button");

    expect(button).toBeInTheDocument();
    expect(button).toMatchSnapshot();
  });
});
