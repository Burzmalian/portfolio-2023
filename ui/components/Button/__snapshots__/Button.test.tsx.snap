// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders a button 1`] = `
<div
  class="bg-slate-400 text-xl text-[red]"
  data-testid="button"
>
  Button
</div>
`;
